DependencyInjection/Compiler/MaybeSetMimeServicesAsAliasesCompilerPass.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * Changes the `liip_imagine.mime_type_guesser` and `liip_imagine.extension_guesser` services to be aliases of the
+ * `mime_types` service provided by the FrameworkBundle when available.
+ *
+ * This compiler pass can be removed when dropping support for Symfony 4.2 and earlier.
+ *
+ * @internal
+ */
+final class MaybeSetMimeServicesAsAliasesCompilerPass extends AbstractCompilerPass
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container): void
+    {
+        if ($container->hasDefinition('mime_types')) {
+            $container->removeDefinition('liip_imagine.mime_type_guesser');
+            $container->removeDefinition('liip_imagine.extension_guesser');
```

Comment:
could this be a BC break if people customized one of these services?

Reply:
Yes, but as is these definitions already are invalid when `symfony/http-foundation:>=5.0` is installed as the `Symfony\Component\HttpFoundation\File\MimeType` namespace just does not exist anymore.  So, there are 3 realistic (IMO) options:

1) This PR, which has a potential B/C break in changing a service definition to a service alias (and, I guess the same B/C concerns as the next item...)

2) Change the service definitions to use the right classes conditionally directly in the container extension class (which, depending on the B/C policy could also be a break because it will change the classes provided by these services and cause existing typehints or instanceof checks to fail if they aren't expecting the counterparts from the `symfony/mime` component)

3) Introduce new services that always point to the new classes and deprecate the existing services, which adds a lot of maintenance burden in having the bundle try to resolve to the correct services based on the active Symfony version, and if anyone is trying to use these services in their own applications they have to go through the effort of finding the right service for their Symfony version

IMO, I'd say this PR is the "best" option moving forward.